src/client/AxiosClient.ts

```diff
@@ -0,0 +1,21 @@
+import axios from "axios";
+import { Configuration } from "../utils/Configuration";
+
+/**
+ * Client to generate instance of Axios that is already configured to call required url
+ */
+export default async () => {
+    const config = Configuration.getInstance();
+    const secretConfig = await config.getSecret();
+
+    if (secretConfig) {
+        return axios.create({
+            baseURL: secretConfig.url,
+            headers: {"x-api-key": secretConfig.key},
+            responseType: "json"
+        });
+    } else {
+        console.log(`Secret details not found.`);
+        return null;
+    }
+};
```

Comment:
Happy to use try/catch and also added the return values, as with the comment above I do like those suggestions.

Can I ask what you see as the benefit of adding the name and doing the traditional function declaration rathan than the arrow function? I presume it was because you changed it from being async but just wondered if there was any other reason?

Reply:
For me personally, when you don't use the function is makes you remove the name in order to make it an anomalous function. Alternatively, you could do some like the following:
`const createAxiosInstance = (): AxiosInstance | null => {
###########
}

export default createAxiosInstance;
`

Reply:
I got rid of the async on this function as I suggested that it be removed from the .getSecret() function. I think it's clearer to see what it is on first glance when you open up the file, with the function name

Reply:
The file itself is called AxiosClient which does give a bit of context as to what it is used for, I think this may be another example of personal preference. Happy to go with the general concencus if any other approvers have any strong opinions either way.